### ISSUE TITLE
Deduce synthetic carry events for event providers which don't annotate them

### ISSUE THREAD

**Issue by DriesDeprest** [PySport/kloppy#379]
Some event providers, e.g. Stats Perform, do not annotate carry events. However, incorporating carry events can be valuable for downstream analysis.

Would it make sense to build some method for an EventDataset that lacks carry events, which contains logic that analyzes the stream of events and insert carry events at the appropriate points within the dataset?

**Comment by probberechts** [comment]
Yes, this could be useful, but based on my experience with SPADL, it is not possible to achieve this with 100% reliability. This could be a helper method and/or option during the deserialization, but I would not do it by default during the deserialization.
